### DISCUSSION TITLE
python-multipart dependency conflicts with multipart require by httpx

### DISCUSSION THREAD

**Discussion by mgorny** [encode/starlette#1960] [Potential Issue] [answered]
httpx has recently gained a dependency on multipart:

```
+multipart==0.2.4
```

However, starlette depends on python-multipart. Both packages install `multipart` (one as `.py` module, the other one as a package), so they both cannot be correctly installed simultaneously.

This is making Gentoo packaging really difficult because a full testing + install of starlette and httpx would require installing both packages simultaneously.

**Comment by Kludex** [comment, marked as answer]
Hmm... On Starlette is an option dependency, on httpx is just used on tests.

@tomchristie can we provide help?

Suggestions besides replacing the multipart on httpx?

**Reply by mgorny** [reply]
Well, I suppose the alternative is to switch to `multipart` in starlette but I don't know pros/cons. From a quick look, both packages are active in git, though python-multipart hasn't seen a release in quite a while.

What really sucks is that both use the same module name while using different pypi names, so this problem is bound to happen over and over again with different packages :-(. That said, I suppose `multipart` wins here since it is the first search result on pypi, while `python-multipart` looks rather meek (late in results, no description…)

**Reply by lovelydinosaur** [reply]
> httpx has recently gained a dependency on multipart

Okay. Let's drop that. https://github.com/encode/httpx/pull/2456

**Reply by mgorny** [reply]
Ok, thanks a lot ;-).
